src/fmu/dataio/dataio.py

```diff
@@ -886,32 +892,32 @@ def generate_metadata(self) -> dict:
             warn(exists_warning, UserWarning)
             return {}
 
-        meta = _metadata.default_meta_dollars()
-        meta["class"] = "case"
-        meta["masterdata"] = _metadata.generate_meta_masterdata(self.config)
-
-        # only asset, not ssdl
-        access = _metadata.generate_meta_access(self.config)
-        assert access is not None
-        meta["access"] = {}
-        meta["access"]["asset"] = access["asset"]
-
-        meta["fmu"] = {}
-        meta["fmu"]["model"] = self.config["model"]
-
-        mcase: Dict[str, Any] = {}  # needed for python < 3.10
-        meta["fmu"]["case"] = mcase
-
-        mcase["name"] = self.casename
-        mcase["uuid"] = str(uuid.uuid4())
-        mcase["user"] = {"id": self.caseuser}
-
-        mcase["description"] = generate_description(self.description)
-
-        meta["tracklog"] = _metadata.generate_meta_tracklog()
-        meta = drop_nones(meta)
+        asset_access = (_metadata.generate_meta_access(self.config) or {}).get("asset")
+
+        meta = CaseSchema(
+            masterdata=Masterdata.model_validate(self.config["masterdata"]),
```

Comment:
I am a bit unsure about this change, the function you replaced it with indicates that the `self.config` can be None and also that `masterdata` might not be in the config. But I guess that might be some relics before we started doing validation of the global configuration file with pydantic?

Reply:
Changed from 
```python
            access=Access(
                asset=Asset.model_validate(
                    (_metadata.generate_meta_access(self.config) or {}).get("asset")
                ),
            )
```

to
```python
access=Access.model_validate(_metadata.generate_meta_access(self.config)),
```